core/libp2p/transport/impl/transport_parser.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TRANSPORT_PARSER_HPP
+#define KAGOME_TRANSPORT_PARSER_HPP
+
+#include <variant>
+
+#include <boost/asio/ip/address.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+
+namespace libp2p::transport {
+
+  /**
+   * Extracts information stored in the provided multiaddress if the protocol
+   * stack is supported by transport implementation
+   */
+  class TransportParser {
+   public:
+    using IpAddress = boost::asio::ip::address;
+    using AddressData = std::variant<std::pair<IpAddress, uint16_t>>;
+
+    enum class Error { PROTOCOLS_UNSUPPORTED = 1 };
+
+    struct ParseResult {
+      /// @brief protocols from the list of supported protocols that were stored
+      /// in the parsed multiaddress
+      const std::list<multi::Protocol::Code> &chosen_protos_;
```

Comment:
What is the rationale of using lists?

Reply:
I guess now there's none in the current version.